Fix screen code naming

mbrukner opened the pull request:
## Checklist

Thanks a lot for your contribution!
Please tick off the following:

- Tests run successfully (i.e. `make test`)
  - [x] using the CC65 compiler
  - [x] using the LLVM/Clang compiler
- [x] [Doxygen](https://www.doxygen.nl/index.html) style tags are used for public API
- [x] Source code is properly formatted; run e.g. `clang-format -i <file>`
- [x] I agree to the [License](https://github.com/MEGA65/mega65-libc/blob/master/LICENSE) of this repository

mbrukner committed:
Fix naming for screen code usage

In the functions that plot actual character to the screen the naming
suggested it would handle PETSCII codes, where in fact is was dealing
with screen codes. These changes make that more clear.
This could be a breaking change for anyone who has been using
fc_plotPetsciiChar in the past.

mbrukner committed:
Fix build instructions when using llvm

mlund commented:
Renaming fcio functions is a breaking change. I suggest to introduce wrapper functions and give a [deprecation warning](https://en.cppreference.com/w/c/language/attributes/deprecated) if the old name is used. Let's also ping @steph72. Please also state if you agree with the new license (#51).

mbrukner committed:
Re-adds the asciiToPetsciicode function and adds a deprecation warning when using LLVM

mbrukner commented:
@mlund makes total sense, I was under the impression with `fcio` only being integrated to the libc with the 0.4.0 release I could get away with just changing the name. That would not work well for people replacing their existing use of fcio with the latest libc though.
I added a deprecation warning as suggested, this unfortunately is not supported in cc65.
I agree releasing under the Apache 2 license.

mlund commented on src/fcio.c:
```diff
@@ -741,7 +749,7 @@ void cr(void)
     }
 }
 
-void fc_plotPetsciiChar(byte x, byte y, byte c, byte color, byte exAttr)
```
Add deprecated wrapper also for `fc_plotPetsciiChar()`.

mlund commented on src/fcio.c:
```diff
@@ -356,7 +356,15 @@ char asciiToPetscii(byte c)
+    return (char)c;
+}
+
+#ifdef __llvm__
```
`__clang__` might be preferred over `__llvm__` as it will then also be recognised by the Calypsi compiler.

mlund replied:
According to the standard it's better to add the deprecation attribute to the function declaration, i.e. `fcio.h`: _"Indicates that the name or entity declared with this attribute is [deprecated](https://en.wikipedia.org/wiki/Deprecation)"_

mbrukner replied:
Good points, while doing that I noticed I was deprecating the wrong function -- this one was never even exposed through the header file. Made some corrections now, taking into account your comments.

mbrukner committed:
Fix deprecation warnings, update doxygen comment, remove deprecation for non-exported function

Removed deprecated asciiToPetsciicode as it was not exported in the
first place. Re-added fc_plotPetsciiChar with deprecation warnings,
updated doxygen comment to reflect these changes.
